src/components/EditCardRow.tsx

```diff
@@ -0,0 +1,51 @@
+import { Button, Input, Td, Tr } from "@chakra-ui/react";
+import { useState } from "react";
+import { server } from "../core/requests";
+import { CardWithStreak } from "../types/card";
+
+interface EditCardRowProps {
+  card: CardWithStreak;
+  onDelete: (id: number) => void;
+}
+
+function EditCardRow({ card, onDelete }: EditCardRowProps) {
+  const [questionInput, setQuestionInput] = useState(card.question);
+  const [answerInput, setAnswerInput] = useState(card.answer);
+
+  const onUpdate = async () => {
+    if (questionInput === "" || answerInput === "") return;
+    await server.patch(`/cards/${card.id}`, {
+      question: questionInput,
+      answer: answerInput,
+    });
+  };
+
+  return (
+    <Tr>
+      <Td>
+        <Input
+          width={"min-content"}
+          variant={"filled"}
+          onChange={(e) => setQuestionInput(e.target.value.slice(0, 255))}
+          value={questionInput}
+        />
+      </Td>
+      <Td>
+        <Input
+          width={"min-content"}
+          variant={"filled"}
+          onChange={(e) => setAnswerInput(e.target.value.slice(0, 255))}
+          value={answerInput}
+        />
+      </Td>
+      <Td>
+        <Button onClick={onUpdate}> UPDATE </Button>
```

Comment:
consider using text transform, screen readers really don't like caps locked stuff, they pronounce it letter by letter